build-tools/automation/azure-pipelines.yaml

```diff
@@ -1323,35 +1325,50 @@ stages:
     - template: yaml-templates/fail-on-issue.yaml
 
 - stage: dotnet_prepare_release
-  displayName: .NET 6 Prepare Release
-  dependsOn: mac_build
+  displayName: Prepare .NET Release
+  dependsOn:
+  - mac_build
+  - linux_build
+  condition: and(eq(dependencies.mac_build.result, 'Succeeded'), eq(dependencies.linux_build.result, 'Succeeded'), eq(variables['MicroBuildSignType'], 'Real'))
   jobs:
-  # Check - "Xamarin.Android (.NET 6 Prepare Release Sign NuGets)"
+  # Check - "Xamarin.Android (Prepare .NET Release Sign Archives)"
   - template: sign-artifacts/jobs/v2.yml@yaml
     parameters:
+      name: sign_net_mac_win
       artifactName: $(NuGetArtifactName)
       signType: $(MicroBuildSignType)
+      signedArtifactName: nuget-signed
+      usePipelineArtifactTasks: true
+
+  # Check - "Xamarin.Android (Prepare .NET Release Sign Linux Archive)"
+  - template: sign-artifacts/jobs/v2.yml@yaml
+    parameters:
+      name: sign_net_linux
+      displayName: Sign Linux Archive
+      artifactName: $(LinuxNuGetArtifactName)
+      signType: $(MicroBuildSignType)
+      signedArtifactName: nuget-linux-signed
       usePipelineArtifactTasks: true
-      condition: eq(variables['MicroBuildSignType'], 'Real')
 
-  # Check - "Xamarin.Android (.NET 6 Prepare Release Convert NuGet to MSI)"
+  # Check - "Xamarin.Android (Prepare .NET Release Convert NuGet to MSI)"
   - template: nuget-msi-convert/job/v2.yml@yaml
     parameters:
       yamlResourceName: yaml
-      dependsOn: signing
+      dependsOn: sign_net_mac_win
       artifactName: nuget-signed
       artifactPatterns: |
         !*Darwin*
       propsArtifactName: nuget-unsigned
       signType: $(MicroBuildSignType)
       runInParallel: false
-      condition: eq(variables['MicroBuildSignType'], 'Real')
 
- # Check - "Xamarin.Android (.NET 6 Prepare Release Push NuGets)"
+  # Check - "Xamarin.Android (Prepare .NET Release Push Internal)"
   - job: push_signed_nugets
-    displayName: Push NuGets
-    dependsOn: nuget_convert
-    condition: and(eq(dependencies.nuget_convert.result, 'Succeeded'), eq(variables['MicroBuildSignType'], 'Real'))
+    displayName: Push Internal
+    dependsOn:
```

Comment:
I'm not *quite* sure how to read the [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema), but shouldn't the value to `dependsOn` be "things that *exist*"? Though I'm not sure how to check "existence", and I don't know where the previously depended on value `nuget_convert` comes from either, so I think I don't understand what this *does*?

Or it doesn't care about "namespaces"?  There does appear to be a `nuget_convert` job: https://github.com/xamarin/yaml-templates/blob/11019011d31d6f5030cddefba088783b4d0ef397/nuget-msi-convert/job/v1.yml#L65

I'm less certain about what the `sign_net_linux` job is, unless template parameter names can provide job names, i.e. `sign_net_linux` is from the `sign-artifacts/jobs/v2.yml@yaml` template invocation above?  (Which would be both awesome and horrifying?)

Reply:
If the `dependsOn` parameter is used on a job, the value should contain one or more job names (not display names) that exist in the same stage of the pipeline. If it is used on a stage, the value should contain one or more stage names. 

The `push_signed_nugets` job has been updated to now depend on both the existing `nuget_convert` and new `sign_net_linux` jobs. As you deduced, the [`nuget_convert job`](https://github.com/xamarin/yaml-templates/blob/11019011d31d6f5030cddefba088783b4d0ef397/nuget-msi-convert/job/v1.yml#L65) is defined in yaml-templates.

YAML templates are quite flexible, and a job name can be specified as a parameter to a job that is declared in a template. This is the case for the shared [`sign-artifacts` job](https://github.com/xamarin/yaml-templates/blob/11019011d31d6f5030cddefba088783b4d0ef397/sign-artifacts/jobs/v2.yml#L33). Since we are now using this signing job template twice in the same stage, I have set the `name` parameter on both instances to ensure that we are using unique job names.